packages/discovery-provider/src/tasks/entity_manager/entities/track.py

```diff
@@ -620,6 +621,38 @@ def delete_track(params: ManageEntityParameters):
     params.add_record(track_id, deleted_track)
 
 
+def download_track(params: ManageEntityParameters):
+    validate_track_tx(params)
+
+    track_id = params.entity_id
+    existing_track = params.existing_records["Track"][track_id]
+    if (
```

Comment:
i don't quite understand this - wouldn't `track_id` never change on an update?
and how is `new_records["Track"][track_id][-1]` an array...
maybe the comment wording is throwing me off.

Reply:
this is all copy-pasted, but I think I understand it?

`track_id` isn't changing, but the track might be (ie if a couple of edits come in the same block). We want to work with the most recent changed track or we might have problems.

`existing_records` is a dict mapping ids => records
`new_records` is a dict mapping ids => record lists